### PR TITLE
Additional documentation on reproducible builds

### LINKED ISSUE #867: 0.18 Document reproducible builds / `cargo risczero build` changes
Update documentation on:

- How Image ID is created and what it promises (and does not promise)
- RISC Zero's definition of reproducible builds are what are our guarantees
- Changes to `cargo risczero build` given new docker based process

### PR DESCRIPTION
**Pull request by heavypackets**
The PR adds deterministic build motivation and additional details on to the risczero build docs, including a link to install the Docker dependency on desktops.

This resolves #867.

### CODE REVIEW

File: risc0/cargo-risczero/README.md [hunk 1]
```diff
@@ -67,15 +69,31 @@ Use the `build-toolchain` command to build the toolchain locally for your host.
...
+```
+
+```bash
 # Build the chess example
 cargo risczero build --manifest-path examples/chess/methods/guest/Cargo.toml
 
+ELFs ready at:
+./elfs/risc0_zkvm_methods_guest/hello_commit - ImageID: eb12f9b97d8759327f651afeb09ae9a5713e7dbc428284d453b8cf56e8dadd5a
+./elfs/risc0_zkvm_methods_guest/multi_test - ImageID: 761900e766a4ae1d8edcb2b49dc9aee54b94e42c9b0d6421cfb112314c4e3efc
+./elfs/risc0_zkvm_methods_guest/slice_io - ImageID: 3f2ad1a2d500ab4ab927eebe241d872d3f598065b8987b182410cf01f350f74c
```
Comment (tzerrell) [inline risc0/cargo-risczero/README.md:98]: Same comment as in https://github.com/risc0/website/pull/272: I don't think these are the chess methods, I think they are the internal/builtin/CI methods. In this context we can probably just drop and show only the factors example
```suggestion
```

File: risc0/cargo-risczero/README.md [hunk 2]
```diff
@@ -67,15 +69,31 @@ Use the `build-toolchain` command to build the toolchain locally for your host.
...
-Use the `build` command to build a guest code for the target `riscv32im-risc0-zkvm-elf`. The resulting ELF binary will be saved on `target/riscv-guest/riscv32im-risc0-zkvm-elf/docker/`. Warning: this requires having docker installed.
+Use the `build` command to build guest code for the zkVM target `riscv32im-risc0-zkvm-elf` deterministically.
+
+The compiled ELF is saved in: `./target/riscv-guest/riscv32im-risc0-zkvm-elf/docker/`
+
+With this containerized build process, we ensure that all builds of your guest code, regardless of the machine or local environment, will produce the same ImageID. The ImageID, and its importance to [security,](https://dev.risczero.com/tech_faq#security) is explained in more detail in our [developer FAQ.](https://dev.risczero.com/tech_faq#zkvm-application-design)
+
+Note: The build command requires the docker CLI installed and in your PATH.
 
 ### Examples
```
Comment (tzerrell) [inline risc0/cargo-risczero/README.md:80]: See other comment -- If we accept that suggestion as-is, we should make this singular.
```suggestion
### Example
```

**Review by tzerrell — COMMENTED** [review]
Looks good up to one comment on the ELFs

File: risc0/cargo-risczero/README.md (hunk 1 above)
Comment (flaub) [reply risc0/cargo-risczero/README.md:98]: Also the output was modified, so now it is ImageID and then the path (which was also modified)

Comment (heavypackets) [reply risc0/cargo-risczero/README.md:98]: I saw the comment on the website FAQ. Will update to reflect @flaub. Thanks!

**Comment by heavypackets** [comment]
Updated to reflect new risczero build output.

File: risc0/cargo-risczero/README.md [hunk 3]
```diff
@@ -67,15 +69,22 @@ Use the `build-toolchain` command to build the toolchain locally for your host.
...
-cargo risczero build --manifest-path examples/factors/methods/guest/Cargo.toml
+With this containerized build process, we ensure that all builds of your guest code, regardless of the machine or local environment, will produce the same ImageID. The ImageID, and its importance to [security,](https://dev.risczero.com/tech_faq#security) is explained in more detail in our [developer FAQ.](https://dev.risczero.com/tech_faq#zkvm-application-design)
+
+Note: The build command requires the docker CLI installed and in your PATH.
 
+### Example
+
+```bash
 # Build the chess example
 cargo risczero build --manifest-path examples/chess/methods/guest/Cargo.toml
```
Comment (flaub) [inline risc0/cargo-risczero/README.md:84]:
```suggestion
cargo risczero build --manifest-path risc0/zkvm/methods/guest/Cargo.toml
```

File: risc0/cargo-risczero/README.md [hunk 4]
```diff
@@ -67,15 +69,22 @@ Use the `build-toolchain` command to build the toolchain locally for your host.
...
-# Build the factors example
-cargo risczero build --manifest-path examples/factors/methods/guest/Cargo.toml
+With this containerized build process, we ensure that all builds of your guest code, regardless of the machine or local environment, will produce the same ImageID. The ImageID, and its importance to [security,](https://dev.risczero.com/tech_faq#security) is explained in more detail in our [developer FAQ.](https://dev.risczero.com/tech_faq#zkvm-application-design)
+
+Note: The build command requires the docker CLI installed and in your PATH.
 
+### Example
+
+```bash
 # Build the chess example
```
Comment (tzerrell) [inline risc0/cargo-risczero/README.md:83]: To couple with @flaub's suggestion:
```suggestion
# Build the zkVM's test examples
```

**Review by tzerrell — APPROVED** [review]

**Pull request merged**